### ISSUE TITLE
Should we introduce a protobuf batch representation?

### ISSUE THREAD

**Issue by jskeet** [cloudevents/spec#794]
It feels like it would be trivial to add a batch representation to https://github.com/cloudevents/spec/blob/v1.0.1/protobuf-format.md, and if we get that done *before* the event format is widely implemented, it's more likely to end up with widespread support.

I'd expect this to literally be a matter of adding a batch content type, and then a message like this:

```proto
message CloudEventBatch {
  repeated CloudEvent events = 1;
}
```

I'm happy to create a PR if that would be useful. (How does that affect versioning of the document within GitHub? It's not clear to me how event formats are versioned...)

**Comment by JemDay** [comment]
+1 - I was thinking about this as well.

**Comment by JemDay** [comment]
@jskeet - Let me know if you're going to raise a PR, if not, I'm happy to do that - should be a quick tweak.

I don't believe versioning will be an issue, technically speaking the batch formats are separate (they have their own media-type) so it's not a breaking change.

**Comment by jskeet** [comment]
@JemDay: If you're happy to do it, that would be great. I don't have a PR ready to go or anything like that. I'm definitely happy to review.

(I'll also be happy to be an early adopter in C#... we're just adding batching into our abstractions, and I'm itching to add the protobuf format as a package too... So I can give feedback if I run into implementation issues.)

**Comment by JemDay** [comment]
@jskeet - OK, I'll try and do something today and link it to this issue.

Good to hear you're looking at C++; the Java support is working but not yet merged into the SDK - that wouldn't support 'batch' today..

**Comment by jskeet** [comment]
@JemDay: Just to clarify, I'm doing C#, not C++. I've no idea what the C++ status is in terms of an SDK :)

**Comment by JemDay** [comment]
@jskeet Typo on my part, for some reason my brain auto-corrected C# -> C++ ;-)

PR Opened for your commentary.

**Comment by jskeet** [comment]
This is now fixed in #801.
